### PR TITLE
logDB provider

### PR DESCRIPTION
**Pull request by hasYuting**
wrap the log storage as a provider
allow uses to config the logDB provider
add the ability to search the logs by message
rewrite the README.md and index.html to get rid of taffydb

Notes: 
search functionality is not very precise (i.e. not allow for substring searching) 
the db is not updated at the point when a new log is inserted

### CODE REVIEW

File: src/storageAdapters/logDB.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+angular.module('fn.logger').provider('logDB', function() {
+  var db = [];
+  var results = [];
+  var self = this;
+
+  /*
+   * set storage configuration
+   */
+
+  this.init = function() {
```
Comment (jackofseattle) [inline src/storageAdapters/logDB.js]: Use `angular.noop` rather than an empty function here.

**Comment by jackofseattle** [comment]
Looks good for a first draft.  Make sure you're adding documentation for how to change the logDB.

**Comment by hasYuting** [comment]
1. As for the documentation, should I also create a demo pages or example pages to show users how to override logDB?

File: README.md [hunk 2]
```diff
@@ -10,25 +10,32 @@ Required bower packages
...
+- Demo can be found at (http://mobileapptracking.github.io/fn-logger/)
 
 Getting started
-----------------
-1. Link scripts:
+---------------
+1. Link scripts
+
 ```
+<script src="../bower_components/jquery/jquery.min.js"></script>
```
Comment (hasdavidc) [inline README.md]: do we still have a jquery dependency?

File: README.md [hunk 3]
```diff
@@ -10,25 +10,32 @@ Required bower packages
...
+<script src="../bower_components/angular/angular.js"></script>
+<script src="../bower_components/angular-sanitize/angular-sanitize.min.js"></script>
+<script src="../bower_components/lodash/dist/lodash.underscore.min.js"></script>
```
Comment (hasdavidc) [inline README.md]: if this could be either lodash or underscore, you can mention that here as well

File: src/config.js [hunk 4]
```diff
@@ -0,0 +1,315 @@
+/**
```
Comment (hasdavidc) [inline src/config.js]: may prefer the name decorators.js for this, as it will be more clear as to what this file is doing

File: src/config.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+angular.module('fn.logger').config(['$provide', 'logDBProvider', function($provide, logDBProvider) {
+  'use strict';
+
+  var $sce;
+  $provide.decorator('$sce', ['$delegate', function($delegate) {
+    return $sce = $delegate;
+  }]);
+
+  $provide.decorator('$log', ['$delegate', function($delegate) {
+    var console = window.console || {};
```
Comment (hasdavidc) [inline src/config.js]: I don't think you wrote this code here (pretty sure this was here before you started working on this), but from here to about the next 30 lines, the code keeps reusing ['error', 'info', 'warn', 'log'] - this should be pulled into its own variable (for example, logLevels) and reused that way. for setting consoleEnabled and dbEnabled, you'll want to set to _.clone(logLevels) though and not just logLevels

Comment (hasdavidc) [inline src/config.js]: it's also used on line 175 and 281 as well

File: src/storageAdapters/logDB.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+   * @return {Array} distinct namespaces
+   */
+
+  this.getNameSpaces = function() {
+    var values = _.pluck(this.db, 'namespace');
+    return _.uniq(values);
+  }
+
+  this.$get = function() {
+    return self;
```
Comment (hasdavidc) [inline src/storageAdapters/logDB.js]: only time you're using self? could probably just `return this` here

File: test/index.html [hunk 7]
```diff
@@ -61,11 +61,10 @@ <h1 class="header">(fn)Logger Test Page</h1>
   <script src="../bower_components/angular/angular.js"></script>
```
Comment (hasdavidc) [inline test/index.html:61]: jQuery above here necessary?

Comment (hasYuting) [inline test/index.html:61]: Yes. we need jQuery in the debugger.js

**Comment by hasYuting** [comment]
Show the inserted log timely under the relevant namespace and level. Even though the logs in this section have been cleared, the new log can still be inserted timely. In the future, user can change the logDB to other database which may have monitors for insert event. Right now, the logDB use a pristine mechanism to guarantee the insert is timely. And instead of adding monitors in debugger.js. controlling the output in logDB provider is simplest way without huge changes on current codes.

File: README.md (hunk 2 above)
Comment (jackofseattle) [inline README.md]: We shouldn't. This would be used for the demo page maybe.

File: README.md (hunk 3 above)
Comment (jackofseattle) [inline README.md]: Correct.
You also don't need angular santize unless you're using > 1.2

**Comment by jackofseattle** [comment]
This is  looking ok.  Please create a section in the documentation showing how to override the logDB.

File: README.md (hunk 2 above)
Comment (hasYuting) [inline README.md]: I saw some code in debugger.js where the jquery is used. in debugger, jquery has been used to set the css and resize the height of the debugger. Should we still skip it?

**Comment by hasdavidc** [comment]
Looks good to me. @jackofseattle - final say?

**Pull request merged**